docs/sovereign/cross-chain-execution.md

```diff
@@ -0,0 +1,53 @@
+
+
+## Cross-Chain Execution within Sovereign Chains
+
+Since a Sovereign Chain is a separate blockchain with a different rule-set from the MultiversX blockchain, there has to be a way of communication between them. The interaction is being done by Smart Contracts, the Sovereign Bridge Service and Nodes. 
+
+![To Sovereign](../../static/sovereign/to-sovereign.png)
+
+When a transaction starts from the MultiversX Mainchain, either from a wallet or a smart contract, it goes through the `Mvx-ESDT-Safe` smart contract. The Observer nodes pay attention to the events that the deposit transaction emits and then the Sovereign Nodes notarize the state changes inside the Sovereign Chain. This notarization means the end of either the Cross-Chain execution or Cross-Chain transfer.
+
+![From Sovereign](../../static/sovereign/from-sovereign.png)
+
+When the transaction sender is from the Sovereign Chain, the `Sov-ESDT-Safe` smart contract must be used. The output of the transaction is read by the Sovereign Bridge Service. From that point there are two possible outcomes:
+1. If the operation is not registered it has to be verified and registered by the `Header-Verifier` smart contract.
+2. If the operation is registered, the Sovereign Bridge Service calls the `Mvx-ESDT-Safe` smart contract deployed on the MultiversX Mainchain to continue the transaction.
+
+This is a high-level description of the whole process, the smart contracts that take place in it are far more detailed and have a lot of specific scenarios and behaviours. The current Sovereign Chain suite consists of four main contracts, here is the high-level description for some of the cross chain smart contracts:
+
+## Mvx-ESDT-Safe & Sov-ESDT-Safe
+The two contracts have the same role: to facilitate a cross-chain execution depending on what side the process starts. The reason for the prefix of `Sov` and `Mvx` is to show where the smart contract is deployed, `Sov` means that the contract is deployed on a Sovereign Chain and `Mvx` that is deployed on the MultiversX Mainchain. There will be an in-depth description of each smart contract in the upcoming modules ([`Mvx-ESDT-Safe`](mvx-esdt-safe.md) and [`Sov-ESDT-Safe`](sov-esdt-safe.md)). The description will consist of flows for the cross-chain interactions, important modules and endpoints. 
+
+## Fee-Market
+Since every Sovereign Chain will have a customizable fee logic, it was paramount that this configuration had to be separated into a different contract. There rules set inside this contract are: 
+* fee per transferred token 
+* fee per gas unit 
+* users whitelist to bypass the fee
+
+This contract is also present in the MultiversX mainchain and in any Sovereign Chain.
```

Comment:
`MultiversX mainchain` - > `MultiversX Mainchain`. Be consistent with the naming across all sections